### ISSUE TITLE
Tests: Fixture implementation duplicated between deprecated `manage/tests/pytest_fixtures.py` and `tools.pytest_fixtures`

### ISSUE THREAD

**Issue by GeigerJ2** [aiidateam/aiida-core#6612]
For instance, `aiida_localhost` in `manage/tests/pytest_fixtures.py`:

https://github.com/aiidateam/aiida-core/blob/dd866ce816e986285f2c5794f431b6e3c68a369b/src/aiida/manage/tests/pytest_fixtures.py#L657-L668

and in `tools/pytest_fixtures/orm.py`:

https://github.com/aiidateam/aiida-core/blob/dd866ce816e986285f2c5794f431b6e3c68a369b/src/aiida/tools/pytest_fixtures/orm.py#L193-L204

with `manage/tests/pytest_fixtures.py` being deprecated:

https://github.com/aiidateam/aiida-core/blob/dd866ce816e986285f2c5794f431b6e3c68a369b/src/aiida/manage/tests/pytest_fixtures.py#L54-L57

Of course, we still need to keep access from the old module for now for backwards compatibility, otherwise plugins break, e.g.:

https://github.com/aiidateam/aiida-quantumespresso/blob/57e7463c5727775d6a0470a41d1aca0ec4083b9a/tests/conftest.py#L13

But rather than duplicating the source code as it seems to be done now, we should let the fixtures in `manage/tests/pytest_fixtures.py` call the actual implementation in `tools/pytest_fixtures/orm.py`. Pinging the `aiida-core` office @agoscinski, @unkcpz, @khsrali.

**Comment by unkcpz** [comment]
I totally agree. I think the doc page https://aiida.readthedocs.io/projects/aiida-core/en/stable/topics/plugins.html#plugin-test-fixtures Is very helpful to show what fixtures that plugin developers can use. We may also want to add same paper for public APIs.

**Comment by danielhollas** [comment]
I honestly don't know if it is worth the churn to be refactoring deprecated functions. I think it is easier to let them be and eventually remove them, and only do new development in the new ones.

**Comment by unkcpz** [comment]
I guess the problem @GeigerJ2 and @khsrali had today was they made changes to new fixtures and then found changes need to apply to the deprecated one as well, correct me if I am wrong. 
The problem is if it is marked as "will be deprecated", do we still need to keep it maintained? The change by pointing to only single definition sounds a good solution for me.

**Comment by agoscinski** [comment]
> I don't have too much experience on how to deprecate things

We don't have a consistent way to do it over the modules but in this case `manage/tests/pytest_fixtures` is anyway not [public API](https://aiida.readthedocs.io/projects/aiida-core/en/stable/reference/api/public.html), so we can just remove it.

**Comment by GeigerJ2** [comment]
> I honestly don't know if it is worth the churn to be refactoring deprecated functions. I think it is easier to let them be and eventually remove them, and only do new development in the new ones.

Yeah, I agree, that's why I only made the issue so far, not a PR :D Though, it should be quite quick. Just remove the source source code there and import from the more recent modules instead.

> I guess the problem @GeigerJ2 and @khsrali had today was they made changes to new fixtures and then found changes need to apply to the deprecated one as well, correct me if I am wrong.

Not really changing anything, but I wanted to look at the fixture implementation, and global search in VS Code yielded two definitions, which confused me.

**Comment by danielhollas** [comment]
> We don't have a consistent way to do it over the modules but in this case manage/tests/pytest_fixtures is anyway not [public API](https://aiida.readthedocs.io/projects/aiida-core/en/stable/reference/api/public.html), so we can just remove it.

Regardless of a definition of a public API, these fixtures are widely used in the AiiDA ecosystem, so we shouldn't remove them just yet IMO, not until a few releases. The new fixtures were only introduced in v2.6 so it'll be a while until everybody catches up.

**Comment by danielhollas** [comment]
> Yeah, I agree, that's why I only made the issue so far, not a PR :D

:+1: :smile: 

> Though, it should be quite quick. Just remove the source code there and import from the more recent modules instead.

To clarify, my main concern was not really about the time to implement it, but the fact that you would be introducing coupling between the `manage` and `tools` module, which I think is not desireable. Also, once you have a single source of truth, if you would change it, it would change the old fixtures as well, which might not be desireable and could break people.
